src/i18n.ts

```diff
@@ -0,0 +1,75 @@
+import i18n from "i18next";
+import { getRuntime } from "@yext/pages/util";
+import {
+  useTranslation as usei18nextTranslation,
+  initReactI18next,
+} from "react-i18next";
+import LanguageDetector from "i18next-browser-languagedetector";
+import type { TemplateRenderProps } from "src/types/entities";
+
+declare global {
+  interface Window {
+    __INITIAL__DATA__: TemplateRenderProps;
+  }
+}
+
+const detector = {
+  name: "pagesJSDetector",
+
+  lookup() {
+    if (getRuntime().name === "browser") {
+      return window.__INITIAL__DATA__.document.locale as string;
+    }
+  },
+
+  cacheUserLanguage() {
+    // Intentionally unimplemented
+  },
+};
+
+const languageDetector = new LanguageDetector();
+languageDetector.addDetector(detector);
+
+const dev = import.meta.env.DEV;
+const lookupOrder = dev ? ["querystring"] : ["pagesJSDetector"];
+
+i18n
+  .use(initReactI18next) // passes i18n down to react-i18next
+  .use(languageDetector)
+  .init({
+    compatibilityJSON: "v3",
+    returnEmptyString: false,
+    detection: {
+      order: lookupOrder,
+      lookupQuerystring: "locale",
+    },
+
+    interpolation: {
+      escapeValue: false, // react already safes from xss
+    },
+  });
+
+export async function getTranslations(localeCode: string) {
+  return await import(`../locales/${localeCode}/translation.json`)
+    .then((module) => ({ [localeCode]: { translation: module.default } }))
+    .catch((_) => {
```

Comment:
Nit, but I believe Typescript prefers having no arguments to using an underscore for an unused argument. At least, that's what the linter I have prefers.
```suggestion
    .catch(() => {
```